### PR TITLE
Change binding type name to camel case in source-gen

### PR DESCRIPTION
**Pull request by satvu**
### Issue describing the changes in this PR

Changes binding type name to camel case in source-gen as customers and partner services may check casing. Consistent with legacy behavior (#589).

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
  * [x] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: sdk/Sdk.Generators/Extensions/StringExtensions.cs [hunk 1]
```diff
@@ -29,5 +30,20 @@ public static string TrimStringsFromEnd(this string str, IReadOnlyList<string> s
...
         }
+
+        /// <summary>
+        /// Returns a copy of the string where the first character is in lower case.
+        /// </summary>
+        public static string ToLowerFirstCharacter(this string str)
+        {
+            if (!string.IsNullOrEmpty(str))
+            {
+                return Char.ToLowerInvariant(str.First()) + str.Substring(1);
```
Comment (jviau) [inline sdk/Sdk.Generators/Extensions/StringExtensions.cs:41]: `str[0]` - slightly less allocation than `First()`

File: sdk/Sdk.Generators/Extensions/StringExtensions.cs [hunk 2]
```diff
@@ -29,5 +30,20 @@ public static string TrimStringsFromEnd(this string str, IReadOnlyList<string> s
 
             return result;
         }
+
+        /// <summary>
+        /// Returns a copy of the string where the first character is in lower case.
+        /// </summary>
+        public static string ToLowerFirstCharacter(this string str)
```
Comment (jviau) [inline sdk/Sdk.Generators/Extensions/StringExtensions.cs:37]: nit:
```suggestion
        public static string ToCamelCase(this string str)
```

**Review by jviau — APPROVED** [review]

**Review by kshyju — APPROVED** [review]

File: sdk/Sdk.Generators/Extensions/StringExtensions.cs (hunk 2 above)
Comment (kshyju) [reply sdk/Sdk.Generators/Extensions/StringExtensions.cs:37]: I am not sure "ToCameCase" is appropriate name as we are not doing that. for example : for input "HTTPTRIGGER" , the ideal camelCase result should be "httpTrigger", but we would return "hTTPTRIGGER"

Comment (satvu) [reply sdk/Sdk.Generators/Extensions/StringExtensions.cs:37]: ^ This was also my reasoning.

Comment (jviau) [reply sdk/Sdk.Generators/Extensions/StringExtensions.cs:37]: Good points

**Pull request merged**